src/modules/blogs/blog.service.ts

```diff
@@ -0,0 +1,129 @@
+import { Blog, blogRepository, userRepository } from "../../common/database";
+import { AppError } from "../../common/errors/app-error";
+import { CreateBlogDto } from "./dto/create-blog.dto";
+import { UpdateBlogDto } from "./dto/update-blog.dto";
+
+export class BlogService {
+  static async create(userId: string, createBlogDto: CreateBlogDto) {
+    try {
+      const user = await userRepository.findOneBy({ id: userId });
+      if (!user) {
+        throw new AppError(400, "User with this id does not exist");
+      }
+      const newBlog = new Blog();
+      newBlog.title = createBlogDto.title;
+      newBlog.content = createBlogDto.content;
+      newBlog.tags = createBlogDto.tags;
+      newBlog.user = user;
+      const blog = await blogRepository.save(newBlog);
+      return { ...blog, user: blog.user.username };
+    } catch (err) {
+      if (err instanceof AppError) {
+        throw err;
+      }
+      throw new AppError(500, "Error creating new blog");
+    }
+  }
+
+  static async getMyBlogs(userId: string, take: number, skip: number) {
+    try {
+      const blogs = await blogRepository.find({
+        where: { user: { id: userId } },
+        relations: ["user"],
+        take: take,
+        skip: skip,
+        order: { created_at: "DESC" },
+      });
+      return blogs.map((blog) => ({
+        ...blog,
+        user: blog.user.username,
+      }));
+    } catch (err) {
+      throw new AppError(500, "Error fetching blogs");
+    }
+  }
+
+  static async getAllBlogs(take: number, skip: number) {
+    try {
+      const blogs = await blogRepository.find({
+        relations: ["user"],
+        take: take,
+        skip: skip,
+        order: { created_at: "DESC" },
+      });
+      return blogs.map((blog) => ({
+        ...blog,
+        user: blog.user.username,
+      }));
+    } catch (err) {
+      throw new AppError(500, "Error fetching blogs");
+    }
+  }
+
+  static async getById(id: string) {
+    try {
+      const blog = await blogRepository.findOne({
+        where: { id },
+        relations: ["user"],
+      });
+      if (!blog) {
+        throw new AppError(404, "Blog not found");
+      }
+      return { ...blog, user: blog.user.username };
```

Comment:
you have relations part in retrieving it retrieves user as nested object. so you don't need to add username as separate field